modules/core/src/test/java/org/apache/ignite/internal/util/VersatileReadWriteLockTest.java

```diff
@@ -277,14 +270,14 @@ void shouldNotAllowInterleavingHoldingReadAndWriteLocks() {
     }
 
     @Test
-    void readLockReleasedLessTimesThanAcquiredShouldStillBeTaken() throws Exception {
+    void readLockReleasedLessTimesThanAcquiredShouldStillBeTaken() {
         lock.readLock();
 
-        Future<?> future = executor.submit(() -> {
+        CompletableFuture<?> future = runAsync(() -> {
             lock.readLock();
             lock.readUnlock();
-        });
-        future.get(10, SECONDS);
+        }, executor);
+        assertThat(future, willCompleteSuccessfully());
```

Comment:
`runWithTimeout` does the same, is that correct? That's why I don't like on-liner methods, you just forget that you could use them

Reply:
Same logic, but different type of a closure